src/core/coap/coap.hpp

```diff
@@ -304,7 +304,8 @@ class EnqueuedResponseHeader
      */
     static void RemoveFrom(Message &aMessage)
     {
-        assert(aMessage.SetLength(aMessage.GetLength() - sizeof(EnqueuedResponseHeader)) == OT_ERROR_NONE);
+        otError error = aMessage.SetLength(aMessage.GetLength() - sizeof(EnqueuedResponseHeader));
+        assert(error == OT_ERROR_NONE);
```

Comment:
Do we need add `OT_UNUSED_VARIABLE(error)`?

Reply:
Thanks, I will add :+1: